Sources/NIO/ByteBuffer-core.swift

```diff
@@ -776,3 +774,14 @@ extension ByteBuffer: Equatable {
         }
     }
 }
+
+extension ByteBuffer {
+    @inlinable
+    func rangeWithinReadableBytesIfReadable(index: Int, length: Int) -> Range<Int>? {
```

Comment:
I think this could just be `rangeWithinReadableBytes`.

Reply:
@Lukasa sounds like a much better name